R/tm_g_association.R

```diff
@@ -222,11 +222,28 @@ srv_tm_g_association <- function(id,
   with_filter <- !missing(filter_panel_api) && inherits(filter_panel_api, "FilterPanelAPI")
   checkmate::assert_class(data, "tdata")
   moduleServer(id, function(input, output, session) {
+
     selector_list <- teal.transform::data_extract_multiple_srv(
       data_extract = list(ref = ref, vars = vars),
-      datasets = data
+      datasets = data,
+      select_validation_rule = list(
+        ref = shinyvalidate::compose_rules(
+          shinyvalidate::sv_required("A reference variable needs to be selected."),
+          ~ if (length(.) != 0 && (.) %in% selector_list()$vars()$select)
+            "Associated variables and reference variable cannot overlap"
+        ),
+        vars = ~ if (length(selector_list()$ref()$select) != 0 && selector_list()$ref()$select %in% (.))
+          "Associated variables and reference variable cannot overlap"
+      )
     )
 
+    iv_r <- reactive({
+      iv <- shinyvalidate::InputValidator$new()
+      iv$add_rule("distribution_theme", shinyvalidate::sv_required("Please select a theme"))
```

Comment:
please make sure that you have full stop everywhere - I assume that's the convention

Reply:
We are using the messages formulated by module authors. Should we intervene?